go/libraries/doltcore/sqle/cluster/controller.go

```diff
@@ -372,38 +390,74 @@ func (c *Controller) RemoteSrvServerArgs(ctx *sql.Context, args remotesrv.Server
 // * Set all databases in database_provider to read-only.
 // * Kill all running queries in GMS.
 // * Replicate all databases to their standby remotes.
-//   - If success, return success.
-//   - If failure, set all databases in database_provider back to their original state. Return failure.
+//   * If success, return success.
+//   * If failure, set all databases in database_provider back to their original state. Return failure.
+//
+// called with c.mu held
 func (c *Controller) gracefulTransitionToStandby() error {
+	c.setProviderIsStandby(true)
+	c.killRunningQueries()
+	// TODO: this can block with c.mu held, although we are not too
+	// interested in the server proceeding gracefully while this is
+	// happening.
+	if err := c.waitForHooksToReplicate(); err != nil {
+		c.setProviderIsStandby(false)
+		c.killRunningQueries()
```

Comment:
why do we call this again here?

Reply:
Because connections could have come in while we were trying to true up the standby and they will have busted InitialDbStates and stuff in their session.